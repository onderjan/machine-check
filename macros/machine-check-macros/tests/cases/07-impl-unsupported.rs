#[machine_check_macros::machine_description]
mod machine {
    struct A {}

    default impl A {}

    unsafe impl A {}

    impl &A {}

    impl A {
        macro_invocation!();

        type AssocType = u32;
        const ASSOC_CONST: u32 = 0;
    }
}

fn main() {}
