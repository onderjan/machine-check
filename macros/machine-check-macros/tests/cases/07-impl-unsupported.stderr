error: machine-check: Defaultness not supported
 --> tests/cases/07-impl-unsupported.rs:5:5
  |
5 |     default impl A {}
  |     ^^^^^^^

error: machine-check: Unsafety not supported
 --> tests/cases/07-impl-unsupported.rs:7:5
  |
7 |     unsafe impl A {}
  |     ^^^^^^

error: machine-check: Non-path type not supported
 --> tests/cases/07-impl-unsupported.rs:9:10
  |
9 |     impl &A {}
  |          ^

error: machine-check: Macro invocations in impl not supported
  --> tests/cases/07-impl-unsupported.rs:12:9
   |
12 |         macro_invocation!();
   |         ^^^^^^^^^^^^^^^^

error: machine-check: Associated consts not supported
  --> tests/cases/07-impl-unsupported.rs:15:9
   |
15 |         const ASSOC_CONST: u32 = 0;
   |         ^^^^^
